applicationset/controllers/applicationset_controller.go

```diff
@@ -456,14 +457,16 @@ func (r *ApplicationSetReconciler) setApplicationSetStatusCondition(ctx context.
 func (r *ApplicationSetReconciler) validateGeneratedApplications(ctx context.Context, desiredApplications []argov1alpha1.Application, applicationSetInfo argov1alpha1.ApplicationSet) (map[int]error, error) {
 	errorsByIndex := map[int]error{}
 	namesSet := map[string]bool{}
+	appProject := &argov1alpha1.AppProject{}
```

Comment:
Are there any risks associated with initializing the appProject outside the loop, such as the k8s client not clearing fields before populating it? I'd worry about, for example, Project A populating the `syncWindows` field and another populating the `sourceRepos` field.

Reply:
Hello :wave: 
(Sorry for delay to answer)

To be honest, I don't know if we can have some risk so to be safe I moved the `appProject` initialization inside the loop :+1: 

PR is up-to-date and rebase on master :smile: 
Do you have any idea when PR can be merged ?